truffle-config.js

```diff
@@ -24,54 +24,45 @@
-   *
-   * $ truffle test --network <network-name>
-   */
+require('dotenv').config();
+const HDWalletProvider = require('truffle-hdwallet-provider-privkey');
+
+const INFURA_PROJECT_ID = process.env.INFURA_PROJECT_ID;
+const DEPLOYMENT_ACCOUNT_PK = process.env.DEPLOYMENT_ACCOUNT_PK;
 
+if (process.env.NODE_ENV !== 'test') {
+  checkEnv();
+}
+
+module.exports = {
   networks: {
-    // Useful for testing. The `development` name is special - truffle uses it by default
-    // if it's defined here and no other network is specified at the command line.
-    // You should run a client (like ganache-cli, geth or parity) in a separate terminal
-    // tab if you use this network and you must also set the `host`, `port` and `network_id`
-    // options below to some value.
-    //
     ganache: {
       host: '127.0.0.1', // Localhost (default: none)
       port: 8545, // Standard Ethereum port (default: none)
       network_id: '*', // Any network (default: none)
     },
-    // Another network with more advanced options...
-    // advanced: {
-    // port: 8777,             // Custom port
-    // network_id: 1342,       // Custom network
-    // gas: 8500000,           // Gas sent with each transaction (default: ~6700000)
-    // gasPrice: 20000000000,  // 20 gwei (in wei) (default: 100 gwei)
-    // from: <address>,        // Account to send txs from (default: accounts[0])
-    // websockets: true        // Enable EventEmitter interface for web3 (default: false)
-    // },
-    // Useful for deploying to a public network.
-    // NB: It's important to wrap the provider as a function.
-    // ropsten: {
-    // provider: () => new HDWalletProvider(mnemonic, `https://ropsten.infura.io/v3/YOUR-PROJECT-ID`),
-    // network_id: 3,       // Ropsten's id
-    // gas: 5500000,        // Ropsten has a lower block limit than mainnet
-    // confirmations: 2,    // # of confs to wait between deployments. (default: 0)
-    // timeoutBlocks: 200,  // # of blocks before a deployment times out  (minimum/default: 50)
-    // skipDryRun: true     // Skip dry run before migrations? (default: false for public nets )
-    // },
-    // Useful for private networks
-    // private: {
-    // provider: () => new HDWalletProvider(mnemonic, `https://network.io`),
-    // network_id: 2111,   // This network is yours, in the cloud.
-    // production: true    // Treats this network as if it was a public net. (default: false)
-    // }
+    ropsten: {
+      provider: () =>
+        new HDWalletProvider(
+          [DEPLOYMENT_ACCOUNT_PK],
+          `https://ropsten.infura.io/v3/${INFURA_PROJECT_ID}`
+        ),
+      network_id: 3,
+      gas: 6700000,
+      gasPrice: 10000000000,
+      skipDryRun: true,
+    },
+    mainnet: {
+      provider: () =>
+        new HDWalletProvider(
+          [DEPLOYMENT_ACCOUNT_PK],
+          `https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`
+        ),
+      network_id: 1,
+      gas: 8000000,
```

Comment:
The gas used to deploy the contract consumes `3213267` of gas. Let's set a closer value like `4000000` for ropsten and mainnet